includes/wikia/Wikia.php

```diff
@@ -1604,18 +1625,9 @@ static public function onSkinTemplateOutputPageBeforeExec(SkinTemplate $skinTemp
 		$tpl->set( 'thisurl', $title->getPrefixedURL() );
 		$tpl->set( 'thisquery', $skinTemplate->thisquery );
 
-		if( !empty( $wgDevelEnvironment ) || !empty( $wgStagingEnvironment ) ) {
-			$out->setRobotPolicy( $wgDefaultRobotPolicy );
-		}
-
-		if( !empty($stagingHeader) ) {
-		// we've got special cases like externaltest.* and showcase.* aliases:
-		// https://github.com/Wikia/wikia-vcl/blob/master/wikia.com/control-stage.vcl#L15
-		// those cases for backend look like production,
-		// therefore we don't want to base only on environment variables
-		// but on HTML headers as well, see:
-		// https://github.com/Wikia/app/blob/dev/redirect-robots.php#L285
-			$out->setRobotPolicy( 'noindex,nofollow' );
+		$robotPolicy = Wikia::getEnvironmentRobotPolicy( $skinTemplate->getRequest() );
+		if ( $robotPolicy ) {
```

Comment:
How about being more strict and use `!is_null()` here? On the other hand there is a nice read: https://www.lucidchart.com/techblog/2015/08/31/the-worst-mistake-of-computer-science/